### ISSUE TITLE
Disable ErrorFlashMessage by default

### ISSUE THREAD

**Issue by neos-bot** [neos/flow-development-collection#578]
Jira issue originally created by user ollehar:

We have a model with a float value, validated with Flow annotations:

```
/****
 * @Flow\Validate(type="NotEmpty")
 * @Flow\Validate(type="Number")
 * @var float
 */
protected $percentage;
```

A controller action new and create. We show flash messages in the layout and validation errors in the template. When validation of float value fails, the error message from the title is displayed as flash message: An error occurred while trying to call %1$s->%2$s()

The validation errors are shown normally.

(Not sure what beta version of Flow we're using, composer.json says "typo3/flow": "3.0.*@beta")

 Jira-URL: https://jira.neos.io/browse/FLOW-322

**Comment by sorenmalling** [comment]
@bwaidelich Based on your work with the flashmessage store, is this still relevant to remove the default error message?

**Comment by bwaidelich** [comment]
This ticket is about making [ActionController::getErrorFlashMessage()](https://github.com/neos/flow-development-collection/blob/7db4d589f3e67024f4863e264d65fc86a86b95e2/Neos.Flow/Classes/Mvc/Controller/ActionController.php#L767-L770) return false by default, so this is not related to the FlashMessage implementation

**Comment by albe** [comment]
Uh, whenever I was about to just do it (which happened more then once 🙈), I always got caught by the thought, what happens when no default error flash message is provided. As in: A new developer, that has not yet customized error messages, what will he see? Then I don't have an answer and immediately stop the attempt.

Should we maybe make it conditional, if the message container is still empty? https://github.com/neos/flow-development-collection/blob/7db4d589f3e67024f4863e264d65fc86a86b95e2/Neos.Flow/Classes/Mvc/Controller/ActionController.php#L709

**Comment by bwaidelich** [comment]
> Should we maybe make it conditional, if the message container is still empty?

Not sure if I got that, could you rephrase the question?

**Comment by albe** [comment]
```php
if (!$this->controllerContext->getFlashMessageContainer()->hasMessages()) {
   $this->controllerContext->getFlashMessageContainer()->addMessage($errorFlashMessage); 
}
```

I'm better with code than with words, so... 😅

**Comment by bwaidelich** [comment]
Thanks for clarifying, a line of code is worth a thousand words :)

We should probably properly rephrase the issue, but IMO the bug is that every developer has to override the `getErrorFlashMessage()` method in their controllers.

Only rendering that weird default message when there are other flash messages present won't really solve that issue IMO.
I would just disable it by default. This is a "breaking" change, but I doubt that anyone actually relies on that default message to be generated..
Theoretically it should be possible to create a core migration to detect that case, but it won't be easy and is probably not worth the effort

**Comment by bwaidelich** [comment]
I removed it from the 7.0 release board since I didn't make it in time.. sorry
